### ISSUE TITLE
Add Attachable.detach()

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#2726]
Many users are unaware of the (rather non-intuitive) possibility of detaching attachables through:

``` java
attachable.attach(null);
```

It may be worth adding a `detach()` method to cover that functionality.
